Added missing permissions for ec2_asg integration tests

mandar242 opened the pull request:
Added missing permissions for ec2_asg integration tests.

tremble reviewed (changes requested):
@mandar242 

There's a limit to the length of these policies which [this change is now hitting](https://dev.azure.com/mattclay/aws-terminator/_build/results?buildId=431&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=5caf77c8-9b10-50ef-b5c7-ca89c63e1c86&l=78):

```
TASK [Fail if policy is too large] *********************************************
fatal: [localhost]: FAILED! => {
    "failed_when_result": true,
    "oversized_policies": [
        "/home/vsts/work/1/s/hacking/aws_config/../../aws/policy/compute.yaml"
    ]
}
```

To avoid this we have a number of Describe* wildcard permissions, which this PR seems to be trying to expand back out.  It also looks like there's a lot of duplicate actions (for example `ec2:AssociateRouteTable` and many of the other routing actions are in Networking.yml.  The way the permissions work it's sufficient for the action to be listed in any of the yml files 


Do you have a failed run that demonstrates the issue you're trying to solve here?

jillr commented on aws/policy/compute.yaml:
```diff
@@ -84,7 +89,9 @@ Statement:
       - ec2:StopInstances
       - ec2:TerminateInstances
-      - ec2:UnassignPrivateIpAddresses
+      - ec2:DescribeLaunchTemplateVersions
```
All ec2 Describe permissions are covered by https://github.com/mattclay/aws-terminator/blob/master/aws/policy/compute.yaml#L74

jillr commented on aws/policy/compute.yaml:
```diff
@@ -174,17 +182,13 @@ Statement:
-      - elasticfilesystem:TagResource
-      - elasticfilesystem:UntagResource
-      - elasticfilesystem:UpdateFileSystem
+      - elasticloadbalancing:DescribeInstanceHealth
```
```suggestion
      - elasticloadbalancing:Describe*
```
To save space in the policies files we prefer to use wildcards on Describe conditions, this way all possible describe conditions are covered by one line

jillr commented on aws/policy/compute.yaml:
```diff
@@ -174,17 +182,13 @@ Statement:
+      - elasticloadbalancing:DescribeLoadBalancers
+      - elasticloadbalancing:DescribeTargetGroupAttributes
+      - elasticloadbalancing:DescribeTargetGroups
+      - sts:GetCallerIdentity
```
What requires GetCallerIdentity here?

mandar242 replied:
This seem to be using ```GetCallerIdentity.```
https://github.com/ansible-collections/community.aws/blob/main/tests/integration/targets/ec2_asg/tasks/main.yml#L93

mandar242 replied:
Removed as discussed.

mandar242 commented:
> @mandar242
> 
> There's a limit to the length of these policies which [this change is now hitting](https://dev.azure.com/mattclay/aws-terminator/_build/results?buildId=431&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=5caf77c8-9b10-50ef-b5c7-ca89c63e1c86&l=78):
> 
> ```
> TASK [Fail if policy is too large] *********************************************
> fatal: [localhost]: FAILED! => {
>     "failed_when_result": true,
>     "oversized_policies": [
>         "/home/vsts/work/1/s/hacking/aws_config/../../aws/policy/compute.yaml"
>     ]
> }
> ```
> 
> To avoid this we have a number of Describe* wildcard permissions, which this PR seems to be trying to expand back out. It also looks like there's a lot of duplicate actions (for example `ec2:AssociateRouteTable` and many of the other routing actions are in Networking.yml). The way the permissions work it's sufficient for the action to be listed in any of the yml files
> 
> Do you have a failed run that demonstrates the issue you're trying to solve here?

@tremble Thank you for feedback. I have removed a few duplicates found, also moved a few actions to correct files. Please feel free to suggest changes if not correct. 

also, I'm trying to run ```ec2_asg``` integration tests, which seem to be failing due to insufficient permission.
example:

```
"msg": "Failed to update autoscaling group: An error occurred (AccessDenied) when calling the DisableMetricsCollection operation
```

tremble commented on aws/policy/compute.yaml:
```diff
@@ -114,7 +118,6 @@ Statement:
     Effect: Allow
     Action:
       - autoscaling:Describe*
-      - ec2:DescribeAvailabilityZones
```
This one's a special case.  Most of our ec2:Describe operations are limited by region, DescribeAvailabilityZones doesn't like being restricted.

tremble commented on aws/policy/compute.yaml:
```diff
@@ -29,6 +29,14 @@ Statement:
       - autoscaling:CancelInstanceRefresh
       - autoscaling:StartInstanceRefresh
       - autoscaling:UpdateAutoScalingGroup
+      - autoscaling:DescribeAutoScalingGroups
+      - autoscaling:DisableMetricsCollection
+      - autoscaling:DescribeLaunchConfigurations
+      - autoscaling:DescribeLifecycleHooks
+      - autoscaling:DetachLoadBalancerTargetGroups
+      - autoscaling:DetachLoadBalancers
+      - autoscaling:EnableMetricsCollection
+      - autoscaling:TerminateInstanceInAutoScalingGroup
```
```suggestion
      - autoscaling:DisableMetricsCollection
      - autoscaling:DetachLoadBalancerTargetGroups
      - autoscaling:DetachLoadBalancers
      - autoscaling:EnableMetricsCollection
      - autoscaling:TerminateInstanceInAutoScalingGroup
```

We have `autoscaling:Describe*` (line 116/120)

jillr reviewed (approved):
Tested locally, LGTM.  The very last assertion in the test fails (at tests/integration/targets/ec2_asg/tasks/main.yml:772) but everything else works, and no permissions failures.  Good work @mandar242!  I'll deploy these to our CI account and we can open a PR to fix the one bad assertion and enable the tests (we'll want to mark them as slow because they are very, very slow)
